### ISSUE TITLE
[V2][Android] Calling mergeOptions leads to Maximum call stack size exceeded

### ISSUE THREAD

**Issue by SergeyBazhanov** [wix/react-native-navigation#5035]
### Issue Description

Hello. Thank you for this awesome package. 
I faced with a problem that sometimes app throws warning `Maximum call stack size exceeded` if i call `mergeOptions` inside my component on Android platform.
Warning is thrown inside this part of code:
```
_createClass(OptionsProcessor, [{
  key: "processOptions",
  value: function processOptions(options) {
    this.processObject(options);
  }
}, {
  key: "processObject",
  value: function processObject(objectToProcess) {
    var _this = this;

    _.forEach(objectToProcess, function (value, key) {
      if (!value) {
        return;
      }

      _this.processComponent(key, value, objectToProcess);

      _this.processColor(key, value, objectToProcess);

      _this.processImage(key, value, objectToProcess);

      _this.processButtonsPassProps(key, value);

      if (!_.isEqual(key, 'passProps') && (_.isObject(value) || _.isArray(value))) {
        _this.processObject(value);
      }
    });
  }
}, {
```
in line `_this.processComponent(key, value, objectToProcess);`

Notice: I've not faced with this warning before i migrated to React Native v0.59.4

Thanks in advance

---
### Environment
* React Native Navigation version: 2.16.0-snapshot.265
* React Native version: 0.59.4
* Platform(s) (iOS, Android, or both?): Android 6.0
* Device info (Simulator/Device? OS version? Debug/Release?): OnePlus One

**Comment by adamrosey** [comment]
@SergeyBazhanov Hey man, did you ever get to the bottom of this? I'm facing the same issue on ios 14 and have been stuck for near a month

**Comment by shakirullahi** [comment]
I'm facing the same issue. any work around for this?

**Comment by shakirullahi** [comment]
@adamrosey @SergeyBazhanov The issue got solved for me. I was using `this` as mentioned official doc, instead of `this.props.componentId` as first parameter of Navigation.mergeOptions(). I don't know why it's wrongly documented here https://wix.github.io/react-native-navigation/docs/sideMenu#opening-via-the-hamburger-menu

**Comment by fiznool** [comment]
Thank you @shakirullahi for this, I faced the same issue. I've opened a PR to fix the docs: #7275
